### ISSUE TITLE
Type Object Error in Tutorial Example

### ISSUE THREAD

**Issue by CoreyRN** [openai/openai-cookbook#71]
Hello,

I've been following the example provided at: https://github.com/openai/openai-cookbook/blob/main/examples/Question_answering_using_embeddings.ipynb

Unfortunately when I get to In [6] I continually obtain the following error in the code:

TypeError                                 Traceback (most recent call last)
[<ipython-input-20-966a87f9b990>](https://localhost:8080/#) in <module>
----> 1 def get_embedding(text: str, model: str=EMBEDDING_MODEL) -> list[float]:
      2     result = openai.Embedding.create(
      3       model=model,
      4       input=text
      5     )

TypeError: 'type' object is not subscriptable

Any recommendations on how to fix this would be much appreciated, as I can't seem to find an answer on Stack Overflow.

Thanks so much for your help!

**Comment by ted-at-openai** [comment]
I'll take a look to see if I can reproduce. I just pushed an update, so it could be something I just introduced.

**Comment by ted-at-openai** [comment]
In the meantime, can you see if running `pip install --upgrade openai` fixes it? Some time ago there was an update to rename engines to models and running an old version of the library sometimes causes errors.

**Comment by ted-at-openai** [comment]
Verified that the code works for me. Which version of Python are you using?

**Comment by ted-at-openai** [comment]
Sounds like it might the list[float] not be understood by a version of Python that's 3.7 or below.

One option is to delete the type hints.

Another option, depending on your version of Python, is to import the types. E.g., `from typing import Float, List` and then use those.

**Comment by CoreyRN** [comment]
Thank you so much for your responses! It's so strange that it is working for you and not me as I tried following the tutorial line by line. I am using python version 3.8.16. I tried both your recommendations but still no luck as I receive the same error. 

When I tried importing the types as you recommended I received the Import Error: cannot import name 'Float' from 'typing' (/usr/lib/python3.8/typing.py).

I have attached an image of my google colab screen demonstrating the error. Any further help would be of much assistance!

Thank you!
![2023-01-11_17-44-05](https://user-images.githubusercontent.com/122490763/211934448-cb740e84-1619-4f6a-a6c8-e92fba9f032e.png)

**Comment by ted-at-openai** [comment]
Thanks for the screenshot and apologies for my insufficient advice. Two amendments:
-  Turns out my memory was a bit off - built-in types were introduced in Python 3.9, not Python 3.8. So makes sense that Python 3.8 would throw this error.
- List can be imported from typing, but not Float

So in Python 3.8, I think the following should work:
```
from typing import List

def get_embedding(text: str, model: str=EMBEDDING_MODEL) -> List[float]:
```

And if it doesn't, try removing the type hints altogether, like:
```
def get_embedding(text: str, model: str=EMBEDDING_MODEL):
```

**Comment by mac4281** [comment]
@ted-at-openai Thanks for your input here.  I'm having the exact same issues with the same example.  I just attempted your fix to no avail.  Still getting: 

TypeError: 'type' object is not subscriptable

Any more thoughts?

**Comment by CoreyRN** [comment]
Thanks so much Ted! With a little ChatGPT assistance I was able to incorporate your responses. So I changed it to:

from typing import List, Tuple, Set, Dict
def get_embedding(text: str, model: str=EMBEDDING_MODEL) -> List[float]:

I also had to change all the other sections such as:

def compute_doc_embeddings(df: pd.DataFrame) -> Dict[Tuple[str, str], List[float]]:

Seems to be working now though! Much appreciated!
